src/main/kotlin/com/hedvig/accountservice/projections/numberFailedCharges/NumberFailedChargesProjection.kt

```diff
@@ -38,14 +44,20 @@ class NumberFailedChargesProjection @Autowired constructor(
     }
 
     @EventHandler
-    fun on(event: ChargeFailedEvent, @Timestamp timestamp: Instant, replayStatus: ReplayStatus) {
+    @Transactional
+    fun on(
+        event: ChargeFailedEvent, @Timestamp timestamp: Instant,
+        replayStatus: ReplayStatus,
+        domainEventMessage: DomainEventMessage<Any>
+    ) {
         val counter = failedChargeCounterManager.incrementNumberFailedCharges(event.memberId, timestamp)
         logger.info {
             "Charge failed: Failed charge number incremented for member=${counter.memberId} to count=${counter.numberFailedCharges}"
         }
         if (!replayStatus.isReplay) {
             notifyAndMaybeTerminateMember(
                 memberId = event.memberId,
+                requestId = domainEventMessage.identifier,
```

Comment:
Smort! I'm wondering if it should be something more unique for this particular project as well?
Like 
```suggestion
                requestId = "${this::class.simpleName}-domainEventMessage.identifier",
```

Reply:
Or something along those lines. If we would want to send events on different reasons than this on an event.

Reply:
Sending two notifications does sound a bit weird but maybe we would want them to be queued or something in the future. I do not know relly.

Reply:
But I think the `domainEventMessage.identifier` is a UUID and that should be unique enough for us :smile:

Reply:
Yea the idea here was if we handle the same event in another location and want to send a notification for a different reason in the service. Those ids would then be the same.

Reply:
I see! yeah let's do that then :)

Reply:
Isn't that kind of unlikely, but if we want to do that let's implement that then